### PR TITLE
Terraform Provider v5 Support

### LINKED ISSUE #70: Dependency Dashboard
This issue lists Renovate updates and detected dependencies. Read the [Dependency Dashboard](https://docs.renovatebot.com/key-concepts/dashboard/) docs to learn more.

## Open

These updates have all been created already. Click a checkbox below to force a retry/rebase of any.

 - [ ] [Update Terraform cloudposse/s3-log-storage/aws to v1.4.3](../pull/95)

## Detected dependencies

<details><summary>terraform</summary>
<blockquote>

<details><summary>examples/complete/main.tf</summary>

</details>

<details><summary>examples/complete/versions.tf</summary>

 - `aws >= 5.0`
 - `local >= 1.2`
 - `hashicorp/terraform >= 1.0`

</details>

<details><summary>main.tf</summary>

 - `cloudposse/label/null 0.25.0`
 - `cloudposse/route53-alias/aws 0.13.0`
 - `cloudposse/s3-log-storage/aws 1.4.2`

</details>

<details><summary>versions.tf</summary>

 - `aws >= 4.9`
 - `local >= 1.2`
 - `hashicorp/terraform >= 1.0`

</details>

</blockquote>
</details>

---

- [ ] Check this box to trigger a request for Renovate to run again on this repository

### LINKED ISSUE #72: Terraform version 0.12 to 13 upgrade issue
here is the module that we are using

module "test" {
  source             = "git::https://github.com/cloudposse/terraform-aws-s3-website.git?ref=tags/0.17.1"
  stage              = local.env
  error_document     = "index.html"
  hostname           = local.my_tasks_web_client_hostname
  index_document     = "index.html"
  parent_zone_name   = local.domain_suffix
  versioning_enabled = true
}

Error: Unsupported attribute
  on .terraform/modules/my_tasks_web_client_with_cname.logs/main.tf line 26, in resource "aws_s3_bucket" "default":
  26:       for_each = var.enable_glacier_transition ? [1] : []
This object does not have an attribute named "enable_glacier_transition".
Error: Unsupported attribute
  on .terraform/modules/my_tasks_web_client_with_cname.logs/main.tf line 40, in resource "aws_s3_bucket" "default":
  40:       for_each = var.enable_glacier_transition ? [1] : []
This object does not have an attribute named "enable_glacier_transition".

### LINKED ISSUE #75: Support AWS Provider v4, Support latest `context.tf`
Have a question? Please checkout our [Slack Community](https://slack.cloudposse.com) or visit our [Slack Archive](https://archive.sweetops.com/). 

[Slack Community](https://slack.cloudposse.com)

## Describe the Feature

* `cloudposse/s3-log-storage/aws` depends on `cloudposse/s3-bucket/aws`, which requires v4 of the AWS Provider
* `context.tf` is still using `null-label` `0.24.1.` instead of `0.25.0`

## Expected Behavior

This module should use the latest version of the aforementioned dependencies

## Use Case

* Latest dependencies should be used
* `tenant` label cannot be used because `null-label` `0.25.0` is not used

## Describe Ideal Solution

Support latest AWS Provider and manually update `context.tf`

## Alternatives Considered

N/A

## Additional Context

N/A

### LINKED ISSUE #83: Failed to query available provider packages when using AWS provider 4.48.0
## Describe the Bug
I am getting the following version error when using `cloudposse/s3-website/aws` version `0.17.3` with:

```hcl
provider "registry.terraform.io/hashicorp/aws" {
  version     = "4.48.0"
....
}
```

```bash
Error: Failed to query available provider packages
Could not retrieve the list of available versions for provider hashicorp/aws: no available releases match the given constraints >= 2.0.0, >= 2.70.0, >= 3.0.0, >= 3.26.0, >= 3.29.0, >= 3.73.0, ~> 4.0, < 4.0.0, >= 4.8.0,
 >= 4.9.0, >= 4.21.0, >= 4.45.0
```

I saw [here](https://github.com/cloudposse/terraform-aws-s3-website/blob/a2b764266a4e736d130a8208b75c3087cff1d385/versions.tf) the restriction was added and [here](https://github.com/cloudposse/terraform-aws-s3-website/blob/86e36f435e0a82f46b12e1b13867898b14893a0e/versions.tf) it should work.

## Steps to Reproduce
1. `terraform init -upgrade` with AWS provider in the latest version

### LINKED ISSUE #84: Changes to default S3 settings cause module to fail
Pardon my improper form, but changes to the default settings for S3 buckets are causing this module to fail:

https://github.com/cloudposse/terraform-aws-s3-website/blob/master/main.tf#L47

https://docs.aws.amazon.com/AmazonS3/latest/userguide/create-bucket-faq.html

'Starting in April 2023, Amazon S3 will change the default settings for S3 Block Public Access and S3 Object Ownership (ACLs disabled) for all new S3 buckets.'

It results in this error:

Error: Error creating S3 bucket: InvalidBucketAclWithObjectOwnership: Bucket cannot have ACLs set with ObjectOwnership's BucketOwnerEnforced setting

### LINKED ISSUE #87: Upgrade to AWS provider 4.x
### Describe the Feature

Upgrade module to use AWS provider 4.x

### Expected Behavior

Module is functional with current version of AWS provider.

### Use Case

Typical usage.

### Describe Ideal Solution

.

### Alternatives Considered

.

### Additional Context

.

### PR DESCRIPTION
**Pull request by milldr**
_**This may be the final PR for this module. This module offers no value beyond [the `s3-bucket` module](https://github.com/cloudposse/terraform-aws-s3-bucket) and [the `terraform-aws-route53-alias` module](https://github.com/cloudposse/terraform-aws-route53-alias), so Cloud Posse is phasing out support for this redundant module. However, this PR should leave this module in a workable state until archival.**_ 

## what
- Terraform v5 updates
- Support for new S3 bucket requirements
- Rebuilt `.github` config
- Apply `tflint` requirements

```
make init
make github/init
make readme
```

## why
- Updates to requirements for CI checks
- Set bucket ownership and grants instead of `public-read` ACL to satisfy new S3 requirements

### Closes Related PRs
- closes #92 
- closes #91
- closes #89 
- closes #88 
- closes #86
- closes #82
- closes #80
- closes #77
- closes #62

### Closes Related Issues
- closes #87
- closes #84
- closes #83
- closes #75
- closes #72
- closes #70

### CODE REVIEW

**Review by Nuru — CHANGES_REQUESTED** [review]
Convert to pass-through to [terraform-aws-s3-bucket](https://github.com/cloudposse/terraform-aws-s3-bucket) so we can archive this module. s3-bucket should already have all the necessary features, and if not, add them there. LMK if it is too much work.

File: examples/complete/versions.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.9"
```
[the remark this answers is not in the document]
Comment (milldr) [reply examples/complete/versions.tf]: added

**Comment by milldr** [comment]
> Convert to pass-through to [terraform-aws-s3-bucket](https://github.com/cloudposse/terraform-aws-s3-bucket) so we can archive this module. s3-bucket should already have all the necessary features, and if not, add them there. LMK if it is too much work.

We decided that we will leave this as a final PR for the module, and redirect users to use the `s3-bucket` module instead. See the README and linked module in this PR description

**Comment by milldr** [comment]
Labeled as a major release to call attention to final PR

**Review by Nuru — APPROVED** [review]

**Pull request merged**